misc/ls-extensions/modules/project-design-service/src/main/java/io/ballerina/projectdesign/generators/GeneratorUtils.java

```diff
@@ -88,4 +99,48 @@ public static ServiceAnnotation getServiceAnnotation(NodeList<AnnotationNode> an
 
         return new ServiceAnnotation(id, label, elementLocation);
     }
+
+    public static ServiceAnnotation getServiceAnnotation(Annotatable annotableSymbol, String filePath) {
+
+        String id = null;
+        String label = "";
+        ElementLocation elementLocation = null;
+
+        List<AnnotationSymbol> annotSymbols = annotableSymbol.annotations();
+        List<AnnotationAttachmentSymbol> annotAttachmentSymbols = annotableSymbol.annotAttachments();
+        if (annotSymbols.size() == annotAttachmentSymbols.size()) {
+            for (int i = 0; i < annotSymbols.size(); i++) {
+                AnnotationSymbol annotSymbol = annotSymbols.get(i);
+                AnnotationAttachmentSymbol annotAttachmentSymbol = annotAttachmentSymbols.get(i);
+                String annotName = annotSymbol.getName().orElse("");
+                elementLocation = annotSymbol.getLocation().isPresent() ?
+                        getElementLocation(filePath, annotSymbol.getLocation().get().lineRange()) : null;
+                if (!(annotName.equals(DISPLAY_ANNOTATION) || annotAttachmentSymbol.attachmentValue().isEmpty()) ||
+                        !(annotAttachmentSymbol.attachmentValue().get().value() instanceof LinkedHashMap)) {
```

Comment:
```suggestion
                        annotAttachmentSymbol.attachmentValue().isPresent() && !(annotAttachmentSymbol.attachmentValue().get().value() instanceof LinkedHashMap)) {
```

Shall we address this possible NPE using `isPresent()` before getting the valus ?

Reply:
I did a minor change to the syntax, as the braces I have already added were wrong. BTW, since I am checking `annotAttachmentSymbol.attachmentValue().isEmpty()` with `||` operator which do short circuit evaluation, there would not be a possibility for NPE right?